source/agora/node/Runner.d

```diff
@@ -201,3 +205,47 @@ private void setVibeLogLevel (ILogger.Level level) @safe
         break;
     }
 }
+
+/*******************************************************************************
+
+    Search multiple paths for SSL certificate and create the TLS context
+
+    Returns:
+        TLS context
+
+*******************************************************************************/
+
+private TLSContext getTLSContext ()
+{
+    const cert_file = "agora-cert.pem";
+    const cert_search_paths = [
+        "./",
+        "/etc/ssl/certs/",
+        "/etc/pki/tls/certs/"
+    ];
+
+    const key_file = "agora-key.pem";
+    const key_search_paths = [
+        "./",
+        "/etc/ssl/private/",
+        "/etc/pki/tls/private/"
+    ];
+
+    TLSContext ctx;
+    foreach (idx; 0 .. cert_search_paths.length)
+    {
+        const cert_path = cert_search_paths[idx] ~ cert_file;
+        const key_path = key_search_paths[idx] ~ key_file;
+
+        // Enable TLS
+        if (exists(cert_path) && exists(key_path))
+        {
+            ctx = createTLSContext(TLSContextKind.server);
+            ctx.useCertificateChainFile(cert_path);
+            ctx.usePrivateKeyFile(key_path);
+            break;
+        }
+    }
```

Comment:
This will only work for both certificate and key file at the same level. Is it intended ?

Reply:
Yep, I don't think we need to search recursively.